src/audio/mux/mux.c

```diff
@@ -269,13 +319,39 @@ static int mux_cmd(struct comp_dev *dev, int cmd, void *data,
 	}
 }
 
+static void prepare_active_look_up(struct comp_dev *dev,
+				   struct audio_stream *sink,
+				   const struct audio_stream **sources)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	const struct audio_stream *source;
+	uint8_t active_elem;
+	uint8_t elem;
+
+	active_elem = 0;
+
+	/* init pointers */
+	for (elem = 0; elem < cd->lookup[0].num_elems; elem++) {
+		source = sources[cd->lookup[0].copy_elem[elem].stream_id];
```

Comment:
lookup[0] ? we should just have one lookup table ?

Reply:
Basically we have one look_up table per sinks. In mux we have one sink, so there is one look up table. In Demux we have several sinks, but we process each sink separately, so we invoke process function only with one look_up table. ```prepare_active_look_up``` is invoked only for mux (one sink).

Yes, @emilchudzik made validation yesterday, and there was a problem with 24 16/16 1ch format. I've made a fix yesterday and we will retrigger validation.